Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs

```diff
@@ -0,0 +1,643 @@
+            }
+            return result;
+        }
+
+        private void AddAnotherClaimButtonClicked(object sender, RoutedEventArgs e)
+        {
+            InputImageViewState = InputViewState.NotSelected;
+            InputFormViewState = InputViewState.NotSelected;
+            this.inputSubmittedGrid.Visibility = Visibility.Collapsed;
+
+            CloseDetailsView();
+        }
+
+        private void DataGridSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            bool isAction = this.dataGrid.Tag != null && this.dataGrid.Tag.ToString() == DataGridAction;
+            if (this.dataGrid.SelectedItem is DataGridViewModel selectedRow && !isAction)
+            {
+                this.detailsViewControl.OpenDetailsView(selectedRow);
+                this.detailsViewControl.Visibility = Visibility.Visible;
+            }
+        }
+
+        private void OnDetailsViewClosed(object sender, EventArgs args)
+        {
+            CloseDetailsView();
+        }
+
+        private void CloseDetailsView()
+        {
+            this.dataGrid.SelectedItem = null;
+            this.detailsViewControl.Visibility = Visibility.Collapsed;
+        }
+
+        private async void OnDeleteDataGridRowButtonClicked(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                this.dataGrid.Tag = DataGridAction;
+
+                var currentRow = (sender as Button)?.DataContext as DataGridViewModel;
+                if (currentRow != null)
+                {
+                    ContentDialog dialog = new ContentDialog
+                    {
+                        Title = "Are you sure?",
+                        Content = $"This operation will delete this claim permanently.",
+                        PrimaryButtonText = "Delete",
+                        SecondaryButtonText = "Cancel",
+                        DefaultButton = ContentDialogButton.Secondary
+                    };
+
+                    ContentDialogResult result = await dialog.ShowAsync();
+                    if (result == ContentDialogResult.Primary)
+                    {
+                        var itemToRemove = DataGridCollection.FirstOrDefault(x => x.Id == currentRow.Id);
+                        bool entryRemovedFromFile = DataGridCollection.Remove(itemToRemove);
+                        if (entryRemovedFromFile)
+                        {
+                            await InsuranceClaimDataLoader.DeleteDataAsync(new List<DataGridViewModel>() { itemToRemove });
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                await Util.GenericApiCallExceptionHandler(ex, "Failure deleting datagrid row");
+            }
+            finally
+            {
+                this.dataGrid.Tag = null;
+                this.dataGrid.SelectedItem = null;
+            }
+        }
+    }
+
+    public enum InputViewState
```

Comment:
Can you move these secondary classes/enums to a DataModel.cs file, under a DataModel folder?

Reply:
Fixed.